### PR TITLE
add unicode WM_SETTINGCHANGE broadcast

### PR DESCRIPTION
**Pull request by lamont-granquist**
helps powershell pick up the env change settings.

### CODE REVIEW

**Comment by lamont-granquist** [comment]
might fix opscode-cookbooks/windows#111

**Comment by lamont-granquist** [comment]
@jdmundrawala this code looks correct but it still doesn't work in test-kitchen

the rubyinstaller also adds its own path already (the windows_path call i had in my recipe isn't executed because of idempotency) but also fails to get added to powershell's path when i switch to a window, so it doesn't look like this is specific to chef any more.

File: lib/chef/mixin/windows_env_helper.rb [hunk 1]
```diff
@@ -39,7 +39,16 @@ module WindowsEnvHelper
...
       def broadcast_env_change
         flags = SMTO_BLOCK | SMTO_ABORTIFHUNG | SMTO_NOTIMEOUTIFNOTHUNG
-        SendMessageTimeoutA(HWND_BROADCAST, WM_SETTINGCHANGE, 0, FFI::MemoryPointer.from_string('Environment').address, flags, 5000, nil)
+        # for why two calls, see:
+        # http://stackoverflow.com/questions/4968373/why-doesnt-sendmessagetimeout-update-the-environment-variables
+        if ( SendMessageTimeoutA(HWND_BROADCAST, WM_SETTINGCHANGE, 0, FFI::MemoryPointer.from_string('Environment').address, flags, 5000, nil) == 0 )
+          Chef::ReservedNames::Win32::Error.raise!
+        end
+        if ( SendMessageTimeoutW(HWND_BROADCAST, WM_SETTINGCHANGE, 0, FFI::MemoryPointer.from_string(
+            Chef::ReservedNames::Win32::Unicode.utf8_to_wide('Environment')
```
Comment (jdmundrawala) [inline lib/chef/mixin/windows_env_helper.rb]: String has a monkey patch that does this:
https://github.com/chef/chef/blob/master/lib/chef/win32/unicode.rb#L53

File: lib/chef/mixin/windows_env_helper.rb [hunk 2]
```diff
@@ -21,11 +21,11 @@
 require 'chef/platform/query_helpers'
 require 'chef/win32/error' if Chef::Platform.windows?
 require 'chef/win32/api/system' if Chef::Platform.windows?
+require 'chef/win32/api/unicode'
```
Comment (jdmundrawala) [inline lib/chef/mixin/windows_env_helper.rb]: Needs guard

**Comment by jdmundrawala** [comment]
I'm :+1: for this. It seems the internet says we should be doing it this way.

File: lib/chef/mixin/windows_env_helper.rb (hunk 1 above)
Comment (lamont-granquist) [inline lib/chef/mixin/windows_env_helper.rb]: yes monkey patching string is vile.

**Pull request merged**